### PR TITLE
initial revision

### PR DESCRIPTION
**Pull request by djdeath**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+uvdefines = [ '-D_GNU_SOURCE', ]
+uvincdir = include_directories('include', 'src', 'src/unix')
+
+if cc.get_define('__ANDROID__') != ''
+  uvsrc += [ 'src/unix/android-ifaddrs.c',
+             'src/unix/pthread-fixes.c' ]
+endif
+
+pthread = cc.find_library('pthread', required: true)
```
Comment (jpakkane) [inline meson.build]: This should be `dependency('threads')` instead. It's more portable and does all the magic extra stuff you need to make threads work.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+libuv_deps = [ cc.find_library('m', required: true),
+               cc.find_library('dl', required: true),
```
Comment (jpakkane) [inline meson.build]: These should both be `required : false`. Most platforms have these inside libc and will fail if these are set to true.

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  'test/test-udp-send-immediate.c',
+  'test/test-udp-send-unreachable.c',
+  'test/test-udp-try-send.c',
+  'test/test-walk-handles.c',
+  'test/test-watcher-cross-stop.c',
+  'test/runner-unix.c',
+]
+
+uvtest = executable('uvtest', testsrc,
+                    dependencies: [libuv_dep, pthread, cc.find_library('util', required: true)])
```
Comment (jpakkane) [inline meson.build]: `required : true` is the default, you don't need to have it here.

**Review by jpakkane — CHANGES_REQUESTED** [review]

**Comment by djdeath** [comment]
Updated thanks!

**Pull request merged**